src/pyOpenMS/tests/unittests/test_AcquisitionInfo.py

```diff
@@ -0,0 +1,67 @@
+import unittest
+import os
+
+import pyopenms
+class TestAcquisitionInfo(unittest.TestCase):
+
+	def setUp(self):
+		dirname = os.path.dirname(os.path.abspath(__file__))
+		self.filename_mzml = os.path.join(dirname, "test.mzML").encode()
+
+	def test_acquisitioninfomemberaccess(self):
+		exp = pyopenms.MSExperiment()
+		pyopenms.MzMLFile().load(self.filename_mzml, exp)
+
+		# Basically test that the output is non-zero (e.g. the data is
+		# correctly relayed to python)
+		# The functionality is not tested here!
+
+		# starting point
+		self.assertEqual(exp[0].getAcquisitionInfo().size(), 1)
+		self.assertNotEqual(exp[0].getAcquisitionInfo().size(), 0)
+
+		# metainfo
+		exp[0].getAcquisitionInfo().size()  # is 1
+		self.assertEqual(exp[0].getAcquisitionInfo()[0].isMetaEmpty(), True)  # is True
+
+                spectra = exp.getSpectra()
+                aqis = spectra[0].getAcquisitionInfo()
+                aqi = aqis[0] # get a copy
+                aqi.setMetaValue('key', 420) # modify it
+                aqis[0] = aqi # and set entry
+                spectra[0].setAcquisitionInfo(aqis)
+                exp.setSpectra(spectra)
+                self.assertEqual(exp[0].getAcquisitionInfo()[0].getMetaValue('key'), 420)  # should be 420
+
+		acin = pyopenms.Acquisition()
+		acin.setMetaValue('key', 42)
+		self.assertEqual(acin.getMetaValue('key'), 42)  # is 42
+		self.assertEqual(acin.isMetaEmpty(), False)  # is False
+
+		# list/vector assignment
+		magicnumber = 3
+		neac = pyopenms.AcquisitionInfo()
+		for i in range(0,magicnumber):
+			neac.push_back(acin)
+
+		self.assertEqual(neac.size(), magicnumber)  # is magicnumber
+
+		# iteration
+		for i in neac:
+			self.assertEqual(i.isMetaEmpty(), False)  # always is False
+		
+		# accession already tested in 2nd section
+
+		# I'd expect assignment to work somewhere
+		exp[0].setAcquisitionInfo(neac)
+		self.assertEqual(exp[0].getAcquisitionInfo().size(), magicnumber)  # should be magicnumber
```

Comment:
fails with_ FAIL: test_acquisitioninfomemberaccess (unittests.test_AcquisitionInfo.TestAcquisitionInfo)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/OpenMS/OpenMS/_build/pyOpenMS/tests/unittests/test_AcquisitionInfo.py", line 57, in test_acquisitioninfomemberaccess
    self.assertEqual(exp[0].getAcquisitionInfo().size(), magicnumber)  # should be magicnumber
AssertionError: 1 != 3

----------------------------------------------------------------------
Ran 293 tests in 30.428s

FAILED (failures=1)

Reply:
@hroest is this because exp[0].setAcquisitionInfo(neac) returns a copy of the first spectrum ?